p2panda-rs/src/document/document.rs

```diff
@@ -0,0 +1,501 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::convert::TryFrom;
+
+use crate::document::{DocumentBuilderError, DocumentView};
+use crate::graph::Graph;
+use crate::hash::Hash;
+use crate::identity::Author;
+use crate::operation::{AsOperation, OperationWithMeta};
+
+/// A resolvable data type made up of a collection of causally linked operations.
+///
+/// Implements the `Resolve` trait for every different schema we support.
+#[derive(Debug, Clone)]
+pub struct Document {
```

Comment:
It could be nice to give some doc strings for each field of the `Document` and `DocumentMeta` struct. You know, they show so nicely up when you generate the Rust Docs :sparkles: :wink:

Reply:
.. OF COURSE with new lines in-between!